File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> m;
        vector<int> v;
        for (int i = 0; i < nums.size(); i++) {
            int numLookingFor = target - nums[i];
            if (m.find(numLookingFor) != m.end()) {
                v.push_back(i);
                v.push_back(m[numLookingFor]);
                return v;
            }
            m[nums[i]] = i;
        }
        return v;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # numNeeded -> index
        m = {}
        for i, num in enumerate(nums):
            numNeeded = target - num
            if numNeeded in m:
                return [i, m[numNeeded]]
            else:
                m[num] = i
        return {}
                